### ISSUE TITLE
dd leaves the cursor on the wrong line and character

### ISSUE THREAD

**Issue by hifi** [nosami/XSVim#164]
Consider this:
```
{
    fo$o
    bar
}
```

When issuing `dd` it should end up like this:
```
{
    $bar
}
```

Instead this happens:
```
{
$   bar
}
```

And when removing a line that has an empty line after it, the cursor incorrectly jumps one line back:
```
{
    fo$o

    bar
}
```

Should do:
```
{
$
    bar
}
```

But it ends up:
```
${

    bar
}
```

Hopefully this explanation was good enough. Used an empty text file in Vim to test for correct behaviour.

I think a general rule is that when a cursor is moved as a result of any line modification command it should skip white space characters and sit at the first non-space character of the line or at the last space character if there are only them on the line.

Same thing happens with the cursor position when you do visual line select and remove lines, the cursor is never positioned correctly after that.

**Comment by nosami** [comment]
Thanks for the detailed report - I'll take a look into this.

**Comment by nosami** [comment]
Ok. This should be fixed in 0.42.2
